test/Microsoft.Diagnostics.EventFlow.Outputs.Tests/EventHubOutputTests.cs

```diff
@@ -55,26 +55,91 @@ public void UsesIsoDateFormat()
+        {
+            var client = new Mock<IEventHubClient>();
+            var healthReporter = new Mock<IHealthReporter>();
+            var configuration = new EventHubOutputConfiguration();
+            configuration.ConnectionString = "Connection string";
+            configuration.EventHubName = "foo";
+            configuration.PartitionKeyProperty = "PartitionByProperty";
+
+            EventData e1 = new EventData();
+            e1.ProviderName = "TestProvider";
+            e1.Timestamp = DateTimeOffset.UtcNow;
+            e1.Level = LogLevel.Warning;
+            e1.Payload.Add("IntProperty", 23);
+            e1.Payload.Add("StringProperty", "partition-perfection");
+            e1.Payload.Add("PartitionByProperty", "partition1");
+
+            EventData e2 = new EventData();
+            e2.ProviderName = "TestProvider";
+            e2.Timestamp = DateTimeOffset.UtcNow;
+            e2.Level = LogLevel.Warning;
+            e2.Payload.Add("IntProperty", 23);
+            e2.Payload.Add("StringProperty", "partition-perfection");
+
+            EventData e3 = new EventData();
+            e3.ProviderName = "TestProvider";
+            e3.Timestamp = DateTimeOffset.UtcNow;
+            e3.Level = LogLevel.Warning;
+            e3.Payload.Add("IntProperty", 23);
+            e3.Payload.Add("StringProperty", new string('a', 150000));
+            e3.Payload.Add("PartitionByProperty", "partition2");
+
+            EventData e4 = new EventData();
+            e4.ProviderName = "TestProvider";
+            e4.Timestamp = DateTimeOffset.UtcNow;
+            e4.Level = LogLevel.Warning;
+            e4.Payload.Add("IntProperty", 23);
+            e4.Payload.Add("StringProperty", new string('b', 150000));
+            e4.Payload.Add("PartitionByProperty", "partition2");
+
+            EventData e5 = new EventData();
+            e5.ProviderName = "TestProvider";
+            e5.Timestamp = DateTimeOffset.UtcNow;
+            e5.Level = LogLevel.Warning;
+            e5.Payload.Add("IntProperty", 23);
+            e5.Payload.Add("StringProperty", new string('c', 150000));
+            e5.Payload.Add("PartitionByProperty", "partition2");
+
+            EventHubOutput eho = new EventHubOutput(configuration, healthReporter.Object, connectionString => client.Object);
+            await eho.SendEventsAsync(new[] {e1, e2, e3, e4, e5}, 17, CancellationToken.None);
+
+            Func<IEnumerable<MessagingEventData>, bool> verifyBatch = batch =>
+            {
+                if (batch.Count() != 1) return false;
+                return true;
+            };
+
+            client.Verify(c => c.SendAsync(It.Is<IEnumerable<MessagingEventData>>(b => verifyBatch(b))), Times.Once);
+            client.Verify(c => c.SendAsync(It.Is<IEnumerable<MessagingEventData>>(b => verifyBatch(b)), "partition1"), Times.Once);
+            client.Verify(c => c.SendAsync(It.Is<IEnumerable<MessagingEventData>>(b => verifyBatch(b)), "partition2"), Times.Exactly(3));
```

Comment:
I would not expect this. I would expect to have one call to SendEventsAsync(batch, "partition2") with 3 elements in the batch.

Reply:
@karolz-ms I setup the three payloads in those events to contain 150k+ bytes of data, so the EventHubMessageSizeLimit kicks in and makes sure these events are outputted to Event Hub in single item batches :)

Reply:
@tedvanderveen ah, OK, that makes more sense. Still, I think it is important to verify that what should be batched is actually batched, because that is a key performance optimization that lets us send thousands of events/sec to EventHub from a single service process. So either this test, or a separate test should verify this.

Reply:
@karolz-ms I shall add an additional test for this!